source/solvers/navier_stokes_base.cc

```diff
@@ -1649,24 +1648,54 @@ NavierStokesBase<dim, VectorType, DofsType>::establish_solid_domain(
                       const unsigned int component =
                         this->fe->system_to_component_index(i).first;
 
-                      // Only pressure DOFs have an additional Dirichlet BC
+                      // Only pressure DOFs have an additional Dirichlet
+                      // condition
                       if (component == dim)
                         {
-                          auto search = dof_is_connected_to_fluid.find(
-                            local_dof_indices[i]);
-                          if (search != dof_is_connected_to_fluid.end())
+                          // We only apply the constraint on the locally owned
+                          // pressure DOFs since we have no way of verifying if
+                          // the locally relevant DOFs are connected to a fluid
+                          // cell
+
+                          bool dof_is_locally_owned = false;
+                          // For the GLS-family of solvers, we only have a
+                          // single index set
+                          if constexpr (std::is_same_v<DofsType, IndexSet>)
                             {
-                              if (non_zero_constraints)
-                                {
-                                  this->nonzero_constraints.add_line(
-                                    local_dof_indices[i]);
-                                  this->nonzero_constraints.set_inhomogeneity(
-                                    local_dof_indices[i], 0);
-                                }
-                              else
+                              dof_is_locally_owned =
+                                locally_owned_dofs.is_element(
+                                  local_dof_indices[i]);
+                            }
+
+                          // For the GD-family of solvers, we have two index
+                          // set. One fore velocities and one for pressure.
+                          if constexpr (std::is_same_v<DofsType,
+                                                       std::vector<IndexSet>>)
+                            {
+                              dof_is_locally_owned =
+                                locally_owned_dofs[1].is_element(
+                                  local_dof_indices[i]);
+                            }
```

Comment:
That is a nice solution to the problem!

Reply:
nice is a big word. Let's say it's a solution? ^^!